sample-apps/react/livestream-app/src/hosts/Hosts.tsx

```diff
@@ -24,28 +21,37 @@ export const characters = [
 const apiKey = import.meta.env.VITE_STREAM_API_KEY as string;
 
 export const Hosts = () => {
+  const [client] = useState<StreamVideoClient>(
+    () => new StreamVideoClient(apiKey),
+  );
+
   const randomCharacter = useMemo(() => {
     const index = Math.floor(Math.random() * characters.length);
     return characters[index];
   }, []);
 
-  const client = useCreateStreamVideoClient({
-    apiKey,
-    tokenOrProvider: async () => {
+  useEffect(() => {
+    const user = {
+      id: randomCharacter,
+      name: randomCharacter,
+      role: 'host',
+    };
+    const tokenProvider = async () => {
       const endpoint = new URL(
         'https://stream-calls-dogfood.vercel.app/api/auth/create-token',
       );
       endpoint.searchParams.set('api_key', apiKey);
       endpoint.searchParams.set('user_id', randomCharacter);
       const response = await fetch(endpoint).then((res) => res.json());
       return response.token as string;
-    },
-    user: {
-      id: randomCharacter,
-      name: randomCharacter,
-      role: 'host',
-    },
-  });
+    };
+
+    client.connectUser(user, tokenProvider);
+
+    return () => {
+      client.disconnectUser();
+    };
+  }, [client]);
```

Comment:
```suggestion
  }, [client, randomCharacter]);
```

I think some dependencies are missing here as well.